include/onnxruntime/core/framework/execution_provider.h

```diff
@@ -259,6 +259,9 @@ class IExecutionProvider {
    */
   virtual int GenerateMetaDefId(const onnxruntime::GraphViewer& graph_viewer, uint64_t& model_hash) const;
 
+  //Generate a unique id for a model. Values are unique for a model instance. 
+  virtual int GenerateModelId(const onnxruntime::GraphViewer& graph_viewer, uint64_t& model_hash) const;
+
```

Comment:
why do we need a separate method for this? we should try to keep the EP interface as simple as possible. 

if the hashing was a standalone function separate from usage of the hash (e.g. using for metadef ids vs model ids) we wouldn't need this would we?

Reply:
The purpose of separating the two id generators is to address pranavsharma's concern that GenerateMetaDefId should serve its own purpose. Or should we get rid of ModelIdGenerator class and move GenerateId to IExecutionProvider level, then each EP can use GenerateId to define its own GenerateMetaDefId or GenerateModelId function? That will require EP code changes.

Reply:
on the topic of keeping EP interface as simple as possible, @pranavsharma and I were discussing whether Generate*Id() should even be a public interface for EP's. It seems to be more of a utility function for implementing GetCapability().